docs/entity_creation.md

```diff
@@ -0,0 +1,203 @@
+
+The graphql schema is amended to include the new entity, with any fields that are required by the future scope of querying. These attributes are preferably only those required inn order to conserve storage.
+```
+0  type TestEntity @entity {
+1  		id: ID!
+2   	entity_field_a: String! @index
+3   	entity_field_b: String! @index
+4   	entity_field_c: String!
+5   	message: Message!
+6   	transaction: Transaction!
+7   	block: Block!
+8  }
+```
+
+- L0 - Entity is initialised with the `@entity` tag
+
+- L1 - Entity is afforded a unique identifier, ID
+
+- L2-4 - Entity attributes and their types are introduced, some with the `@index` tag to allow more complex querying
+
+- L5-8 - Relevant message, transaction and block are provided to the entity
+
+Updating the system to include these changes can be done by running the following: 
+```
+yarn codegen
+```
+
+
+#### Implementing the entity handler
+
+in `/src/mappings/your_entity_category/entity_name.ts`
+
+This handler function will be configured to trigger whenever the indexer detects the desired interaction with the entity:
+
+```
+0	export async function handleYourEntityName(event: CosmosEvent): Promise<void> {
+1		logger.info(`confirmation that your handler is triggered`);
+2
+3		const id = messageId(event.msg);
+4  		const decoded_msg = event.msg.msg.decodedMsg;
+5  		const entity_field_a = decoded_msg.entity_field_a;
+6  		const entity_field_b = decoded_msg.entity_field_b;
+7  		const entity_field_c = decoded_msg.entity_field_c;
+8
+9  		if (!entity_field_a || !entity_field_b || !entity_field_c) {
+10    		logger.warn(`indexing failed ... debug info`)
+11    		return
+12  	}
+13
+14    	const entity = TestEntity.create({
+15    		id,
+16    		entity_field_a,
+17    		entity_field_b,
+18    		entity_field_c,
+19    		messageId: id,
+20    		transactionId: event.msg.tx.hash,
+21    		blockId: event.msg.block.block.id
+22    	});
+23	
+24		await entity.save();
+25	}
+```
+
+- L0 - Create an synchronous handler function accepting a `CosmosEvent` as a parameter
+
+- L4-7 - Create a reference to the message relevant to the incoming Event and destructure the message into the desired fields
+
+- L9-12 - Check that these fields have been destructured correctly
+
+- L14-22 - Create a `TestEntity` instance and populate the required fields
+
+- L24 - Save the entity to the database
+
+#### Configuring handler filters
+
+in `project.yaml`:
+
+This example will run the previously instantiated handler whenever the filtering conditions are met - e.g. when an event of type `"event_type"` is captured as triggered by a message with the type `"message_type_url"`
+
+```
+0  - handler: handleYourEntityName
+1    kind: cosmos/EventHandler
+2    filter:
+3    	type: "event_type"
+4        messageFilter:
+5        	type: "message_type_url" 
+```
+
+- L0 - Reference the appropriate handler to be called when triggered
+
+- L1 - Define which type of handler this will trigger, in this case an Event handler
```

Comment:
Can we just elaborate a little bit to explain that this impacts the handler function signature. Maybe there's a subquery docs link relevant to this as well.